include/mapbox/variant_visitor.hpp

```diff
@@ -13,7 +15,7 @@ struct visitor<Fn> : Fn
     using type = Fn;
     using Fn::operator();
 
-    visitor(Fn fn) : Fn(fn) {}
+    visitor(Fn&& fn) : Fn(std::forward<Fn>(fn)) {}
```

Comment:
Nice, it's simpler this way. Pull request has been updated. 9f991da78d3146d32be67695a89c2b1197c826b2